### PR TITLE
gergo/pre commit

### PR DESCRIPTION
**Pull request by gjedlicska**
- Adds hadolint as pre-commit
- Addresses all hadolint comments
- Hadolint docker works when entrypoint explicitly provided
- Use noninteractive apt-get frontend and clean after install
- build(circleci): filters for pre-commit should be same as for test-server
- build(circleci): enable remote docker for pre-commit
- build(circleci): use speckle pre-commit runner with built-in hadolint
- build(server): dockerfile RUN statements are consolidated
- build(server): dockerfile RUN statements are consolidated
- Improve husky bash script to catch errors
- Integrates pre-commit with husky
- Update README for revised developer instructions
- Updates pre-commit yarn script
- refactor(pre-commit): make everyone happy with loosly integrating husky and pre-commit scripts

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -213,7 +213,7 @@ jobs:
             - ~/.cache/pre-commit
       - run:
           name: Run pre-commit
-          command: pre-commit run --all-files --config <<parameters.config_file>>
+          command: ./.husky/pre-commit
       - run:
           name: Run deployment pre-commit
```
Comment (iainsproat) [inline .circleci/config.yml:218]: Can we remove the second pre-commit file & CI step now?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,29 +1,9 @@
...
-          - '@typescript-eslint/parser@5.21.0'
-          - typescript@4.5.4
-          - '@rushstack/eslint-patch@1.1.3'
-          - '@vue/eslint-config-typescript@11.0.0'
-
-  - repo: https://github.com/pre-commit/mirrors-prettier
-    rev: 'v2.7.1' # Use the sha / tag you want to point at
-    hooks:
-      - id: prettier
+      - id: hadolint
```
Comment (iainsproat) [inline .pre-commit-config.yaml:7]: I think we also need shellcheck & helm lint from the other pre-commit config file.

Comment (gjedlicska) [reply .pre-commit-config.yaml:7]: yup, ur right

File: .circleci/config.yml (hunk 1 above)
Comment (gjedlicska) [reply .circleci/config.yml:218]: :y

**Review by iainsproat — APPROVED** [review]

**Pull request merged**